pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,385 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from __future__ import print_function
+
+import numpy as np
+import scipy.constants
+from pyiron.atomistics.master.parallel import AtomisticParallelMaster
+from pyiron_base import JobGenerator
+from sklearn.linear_model import LinearRegression
+from collections import defaultdict
+import warnings
+import itertools
+
+__author__ = "Joerg Neugebauer, Jan Janssen"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Jan Janssen"
+__email__ = "janssen@mpie.de"
+__status__ = "production"
+__date__ = "Oct 21, 2020"
+
+
+eV_div_A3_to_GPa = (
+    1e21 / scipy.constants.physical_constants["joule-electron volt relationship"][0]
+)
+
+def _fit_coeffs_with_stress(
+        strain,
+        stress,
+        rotations,
+        max_polynomial_order=None,
+        fit_first_order=False,
+    ):
+    higher_terms = _get_higher_order_terms(
+        strain,
+        max_polynomial_order=max_polynomial_order,
+        rotations=rotations,
+        derivative=True,
+    )
+    strain = np.einsum('nik,mkl,njl->nmij',
```

Comment:
Twice here you take an einsum and then re-order the output with a magic map `[0, 4, 8, 5, 2, 1]`. I think it would benefit from being factored out as a function with a good name, because right now it seems very magical.